imsApp/src/imsRecord.cc

```diff
@@ -3589,6 +3647,142 @@ static void post_fields( imsRecord *prec, unsigned short alarm_mask,
     return;
 }
 
+/******************************************************************************/
+static void io_controller ( struct ims_info *mInfo )
+{
+    char *s, *t, buf[2048];
+    const double wtimeout = 1.0;
+    const double rtimeout = 0.1;
+    asynStatus status;
+    int i, eomReason, wild;
+    size_t len;
+
+    for (;;) {
+	/* Check for new work */
+	mInfo->rMutex->lock();
+	for (i = 0; i < REQ_CNT; i++) {
+	    struct ims_asyn_req *req = &mInfo->req[i];
+
+	    if (req->state == IAR_SENDONLY) {
+		pasynOctetSyncIO->write( mInfo->pasynUser, req->out, 
+					 req->outsize, wtimeout, &len );
+		free_request_buf(mInfo, req);
+		continue;
+	    }
+	    if (req->state == IAR_SEND) {
+		pasynOctetSyncIO->write( mInfo->pasynUser, req->out, 
+					 req->outsize, wtimeout, &len );
+		req->state = IAR_RESPWAIT;
+		continue;
+	    }
+	}
+	mInfo->rMutex->unlock();
+
+	/* Try to read a message */
+	status = pasynOctetSyncIO->read( mInfo->pasynUser, buf, sizeof(buf) - 1,
+					 rtimeout, &len, &eomReason );
+	if (status != asynSuccess || len <= 0)
+	    continue;
+
+	if (len < sizeof(buf))
+	    buf[len] = 0;
+
+#if 0
+	/* Debug print. */
+	printf("< ");
+	for (size_t ii = 0; ii < len; ii++) {
+	    if (buf[ii] == '\r')
+		printf("\\r");
+	    else if (buf[ii] == '\n')
+		printf("\\n");
+	    else
+		putchar(buf[ii]);
+	}
+	putchar('\n');
+#endif
+
+	/* Is it even possible to have more than one message?!? */
+	for (s = buf, t = index(s, '\r'); s && *s; s = t) {
```

Comment:
Because this is new code, a slightly more informative comment would be appreaciative so future devs can take a look at this line when debugging something totally unrelated and know what you are getting across

Reply:
I can clarify.  But before I edit the comment, I'm not sure that the read will *ever* give me two messages at once.  (Perhaps the kernel read does, but I tend to doubt that.  But the asyn read almost certainly checks for the EOS characters.)

Paranoia.